### ISSUE TITLE
Disable or remove Google Drive projects

### ISSUE THREAD

**Issue by seadowg** [getodk/collect#5402]
We want to remove Google Drive projects that have no unsent forms and "disable" ones that have unsent forms.

## Acceptance

- [x] Given I've created a Google Drive project in an earlier version of Collect
And it's set as my current project
And I've also created a ODK server project
When I upgrade to and launch v2023.3
Then my current project changes to the ODK server project
And the Google Drive project is gone (in the UI and on disk)

- [x] Given I've created a Google Drive project in an earlier version of Collect
And it's set as my current project
And I've not created any other projects
When I upgrade to and launch v2023.3
Then I see the first launch screen
And the Google Drive project is gone (in the UI and on disk)

- [x] Given I've created a Google Drive project in an earlier version of Collect
And I have unsent submissions (finalized or non finalized)
When I upgrade to and launch v2023.3
Then the Google Drive project is still there
And the starting new forms should be blocked (there should be a dialog displayed after trying to fill a new form)
And there is a banner with a warning about the project being deleted in the next release (text TBD)

- [x] Given I've created a Google Drive project in an earlier version of Collect
And I have unsent submissions (finalized or non finalized)
When I upgrade to and launch v2023.3
And I click learn more on the warning banner (taking me to the Google Drive deprecation announcement)
And I return to the app
Then the warning banner is still there

**Comment by seadowg** [comment]
After writing this out, I've realized that we have a blind spot: what happens to Google Drive projects with drafts (unfinalized instances)? We're planning to get finalized instances off the device from "hidden" projects with #5401, but this would not take care of the drafts.

My proposal is that we keep projects with drafts still available, but disable "Start Blank Form" and still hide them when the drafts are all finalized or deleted. It'd be good to get input here from @grzesiek2010, @alyblenkin and also @lognaturel (as anything we do here will be controversial).

**Comment by grzesiek2010** [comment]
I started thinking if it makes sense to be that careful at all... it looks like our users have listened to us that they shouldn't update the app if they want to continue using Google Drive integration because according to Firebase we have had only 2 such users in the latest v2023.1.x version:

![image](https://github.com/getodk/collect/assets/3276264/7b1064a2-b6dd-4f5d-b549-9425a44921dd)

Maybe in v2023.3 we can just remove GD projects with no saved forms and then in v2023.4 remove the GD integration at all if the number of users is still that small? What do you think?

**Comment by seadowg** [comment]
Ah, that's good to know! I'm guessing a lot of folks stayed on the older version (without the Google Drive projects warning banner), but it could just be that some others have been slow to upgrade. Looking back at [our announcement](https://forum.getodk.org/t/collect-will-stop-connecting-to-google-drive-sheets-in-fall-2023/40097), we did say that we'd remove all Google Drive features in v2023.3, but I personally still feel a little uncomfortable deleting unsent instances - I'd still prefer to remove access to the projects in v2023.3 and then completely delete everything in v2023.4. This gives people a chance to get data off the phone once they can't access it through the UI any longer. I think we still need to do #5401 as well, as it's very complicated to pull the data from disk and then get it into a Google Sheet.

That's my take, but others might think I'm being overcautious!

**Comment by alyblenkin** [comment]
It's great to see the analytics and that some people are following our recommendation. 

If it's not going to slow us down, I think leaning on the side of caution is a good call. We want to avoid disrupting workflows, even if it's a small group of users. I think removing the ability to "Start Blank Form", keep drafts, and then hide them when drafts are finalized or deleted makes sense.

We could send another follow-up announcement before we delete everything in v2023.4 to ensure people don't lose and unsent instances. If we are really concerned, we could also introduce the blue banner/warning message to make sure users take action, but that might be too much work.

**Comment by grzesiek2010** [comment]
I agree with the two first acceptance criteria but then we have:

```
Given I've created a Google Drive project in an earlier version of Collect
And I have finalized but unsent forms
When I upgrade to and launch v2023.3
Then the Google Drive project is gone
And the unsent forms are still available on disk
```

Maybe we should block starting new forms but still allow uploading the finalized ones by not hiding that project? It would be similar to the scenario with drafts. To be honest given that the number of GD users is so small I don't think there is a point in distinguishing those two cases.

and the last case:

```
Given I've created a Google Drive project in an earlier version of Collect
And I have drafts
When I upgrade to and launch v2023.3
And I finalize the drafts
And I restart the app
Then the Google Drive project is gone
And the unsent forms are still available on disk
```

Here again, I think we should allow uploading the data and only block starting new forms.

**Comment by seadowg** [comment]
@grzesiek2010 I think I agree. You'll see I have some questions about adding forced auto send (in #5401), I think going with your suggestion of just hiding "Start new form" (rather than removing/hiding the project) for all Google Drive projects with unsent forms and then also not doing #5401 makes sense. We'd still remove Google Drive projects with only sent (or no) forms. Sound good @alyblenkin?

**Comment by alyblenkin** [comment]
I think this makes sense! I don't have a lot of context on this piece of work beyond the announcement that was made in Jan, so I may be overthinking this, but I'm still wondering how we inform the user about what will happen after they upgrade. I'm specifically thinking about our principle around giving users visibility into what's happening, and what happened so there aren't any surprises or confusion.

Do we have the ability to show a confirmation dialog when they go to upgrade? I think we should communicate what will happen next (i.e. that their Google Drive project will be removed and all sent forms, but they can still access the unsent forms until v2023.4). Or if we decide to hide everything, we should make that very clear to the users just in case they've missed the forum post.

**Comment by alyblenkin** [comment]
Update: @grzesiek2010 I just had a quick chat with Callum about some of the context around this issue, and I have a better understanding of your proposal, and it makes sense. 

So for the Google Drive projects, we take away the ability to start new forms. The only thing I would add is to continue showing a warning banner to communicate what will happen next. We spoke about the banner copy being updated to say something like: 

"Your forms will be deleted in the next version of Collect. If you need these forms, get them off the device." Or link directly to the announcement.  

As you pointed out @ grzesiek2010, this is a tiny group of people because most people have shifted away, but we should show the banner to avoid surprises. 

@seadowg - jump in if I missed anything!

**Comment by seadowg** [comment]
I've updated the acceptance to reflect the discussion. I think we can iterate on the text when we have a PR for this. One thing to note is that I've specified that the banner should always show now (rather than hide when the user clicks "Learn more"). As we're disabling the ability to start new forms, I'm less worried about disrupting data collection and more concerned with pestering the user to take care of their remaining unsent forms.

I'm going to say this is ready to go now, and I'll also close #5401. @lognaturel feel free to jump in if you feel uncomfortable with any of this.
